components/script/dom/fakexrdevice.rs

```diff
@@ -7,43 +7,39 @@ use crate::dom::bindings::codegen::Bindings::FakeXRDeviceBinding::{
 };
 use crate::dom::bindings::codegen::Bindings::XRViewBinding::XREye;
 use crate::dom::bindings::error::{Error, Fallible};
-use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::reflector::{reflect_dom_object, Reflector};
 use crate::dom::bindings::root::DomRoot;
 use crate::dom::globalscope::GlobalScope;
 use dom_struct::dom_struct;
-use webvr_traits::{MockVRControlMsg, MockVRView, WebVRMsg};
+use euclid::{TypedRigidTransform3D, TypedRotation3D, TypedTransform3D, TypedVector3D};
+use ipc_channel::ipc::IpcSender;
+use webxr_api::{MockDeviceMsg, View, Views};
 
 #[dom_struct]
 pub struct FakeXRDevice {
     reflector: Reflector,
+    #[ignore_malloc_size_of = "defined in ipc-channel"]
+    sender: IpcSender<MockDeviceMsg>,
 }
 
 impl FakeXRDevice {
-    pub fn new_inherited() -> FakeXRDevice {
+    pub fn new_inherited(sender: IpcSender<MockDeviceMsg>) -> FakeXRDevice {
         FakeXRDevice {
             reflector: Reflector::new(),
+            sender,
         }
     }
 
-    pub fn new(global: &GlobalScope) -> DomRoot<FakeXRDevice> {
+    pub fn new(global: &GlobalScope, sender: IpcSender<MockDeviceMsg>) -> DomRoot<FakeXRDevice> {
         reflect_dom_object(
-            Box::new(FakeXRDevice::new_inherited()),
+            Box::new(FakeXRDevice::new_inherited(sender)),
             global,
             FakeXRDeviceBinding::Wrap,
         )
     }
-
-    fn send_msg(&self, msg: MockVRControlMsg) {
-        self.global()
-            .as_window()
-            .webvr_thread()
-            .unwrap()
-            .send(WebVRMsg::MessageMockDisplay(msg))
-            .unwrap();
-    }
 }
 
-pub fn get_views(views: &[FakeXRViewInit]) -> Fallible<(MockVRView, MockVRView)> {
+pub fn get_views(views: &[FakeXRViewInit]) -> Fallible<Views> {
     if views.len() != 2 {
         return Err(Error::NotSupported);
```

Comment:
Might want to support mono at some point.

Reply:
Yeah, inline support needs a bit more work